src/main/java/com/FlightPub/Controllers/IndexController.java

```diff
@@ -218,12 +219,12 @@ public String runSearch(@ModelAttribute BasicSearch search, Model model, HttpSes
     @PostMapping("/advancedSearch")
     public String runAdvancedSearch(@ModelAttribute BasicSearch search, Model model, HttpSession session) {
         model = addDateAndTimeToModel(model);
-        List<Flight> flights;
+        List<Flight>[] flights = new ArrayList[2];
         List<StopOver>[] stopOver = new ArrayList[3];
         search.setFlightServices(flightServices);
         search.setLocationServices(locationServices);
         try{
-            flights =  search.runAdvancedSearch(this.getSession(session).getUsr());
+            flights[0] =  search.runAdvancedSearch(this.getSession(session).getUsr());
             if(!search.isDirectFlight()) {
                 stopOver[0] = search.advancedStopOverSearch(this.getSession(session).getUsr(), 1);
```

Comment:
should we move the model attribute for stop overs to inside this logic to minimize the amount of objects passed to the page?